pkg/pyramid/tier_fs.go

```diff
@@ -114,7 +115,12 @@ func (tfs *TierFS) removeFromLocal(rPath params.RelativePath, filesize int64) {
 	// This will be called by the cache eviction mechanism during entry insert.
 	// We don't want to wait while the file is being removed from the local disk.
 	evictionHistograms.WithLabelValues(tfs.fsName).Observe(float64(filesize))
-	go tfs.removeFromLocalInternal(rPath)
+	// Decrement the reference count on the file. When the ref count reaches -1, The file will be deleted
+	tfs.fileTracker.dec(rPath, func(ref int) {
```

Comment:
I think we need a 'delete' marker on the map - we like that for every time we called open we inc/dec in case the file wasn't found - it means that evict delete it.
if we like a better visibility I think the tracker should understand when to perform the callback or make sure that every time you call dec you also check and delete the file.

Something like:
tracker := NewTracker(deleteCallback) // callback to the implementation on how to delete file from local

closer := tracker.Open(path); // return a callback 
defer closer() // closer will take the lock remove the ref count when 0 and remove from map will check if a delete flag is mark and call the callback

tracker.Delete(path) // mark the file for delete or delete by keeping the lock